blog/_posts/2015-09-29-compose3d-threejs.md

```diff
@@ -0,0 +1,192 @@
+---
+
+Over the last three months, I've been working on [Compose3D](https://github.com/rohitvarkey/Compose3D.jl),
+which is an extension of the amazing [Compose](https://github.com/dcjones/Compose.jl) package to 3D. My work on
+Compose3D began as a project for my Computer Graphics course along with [Pranav T Bhat](https://github.com/pranavtbhat),
+and by the end of the course, we had a working prototype for Compose3D with support for contexts and geometries and a
+very basic WebGL backend.
+
+It has been my pleasure to have been able to continue this work under the guidance of [Shashi Gowda](https://github.com/shashi)
+and [Simon Danisch](https://github.com/SimonDanisch) as a part of the first ever Julia Summer of Code, generously
+sponsored by the [Gordon and Betty Moore Foundation](https://www.moore.org/). While I've been able to add quite a lot of
+functionality to Compose3D, it isn't totally ready for release yet. Hopefully, in some time it
+will be ready. But as a happy side effect, I have been able to abstract out the WebGL rendering functionality provided
+by the original prototype (and a lot more!) to a separate package called
+[ThreeJS.jl](https://github.com/rohitvarkey/ThreeJS.jl),
+which can now be used to render 3D graphics in browsers using Julia, opening up possibilities of displaying such
+scenes in [IJulia](https://github.com/JuliaLang/IJulia.jl) notebooks and [Escher](https://github.com/shashi/Escher.jl).
+
+## ThreeJS.jl
+
+ThreeJS is now responsible for all the WebGL rendering done by Compose3D. It can also be used as a standalone package for
+other graphics packages to use as a backend.
+
+Initially, my approach to render scenes in Compose3D was to just emit out the corresponding JavaScript code, into the
+IJulia notebook, which would then run it! This worked pretty well in IJulia notebooks, but it was soon apparent that
+there were several flaws with this approach.
+
+  - It was hard to extend.
+  - Did not play well with Escher.
+  - Nor did it work with Interact to provide interactivity.
+
+So Shashi suggested implementing a [Polymer](https://www.polymer-project.org/1.0/) wrapper around the excellent
+[three.js](http://threejs.org/) library, to create threejs web components. The Polymer team had done some work on
+creating threejs components and had a basic implementation of the same ready, which I promptly [forked](https://github.com/rohitvarkey/three-js)
+and tweaked to add functionality I needed. It's quite safe to say that I've spent more time writing JavaScript than
+Julia during JSoC!
+
+Switching over to using web components suddenly opened up 2 major avenues. Compose3D could now work with Escher and
+also provided interactivity. Shashi demoed this during his talk at The Fifth Elephant with a Sierpinski pyramid
+whose number of levels of recursion could be changed using a slider. ThreeJS outputs [Patchwork](https://github.com/shashi/Patchwork.jl)
```

Comment:
this line about me demoing adds no value, you can put a small snippet after your sierpinski code below, showing how the sierpinski function can be connected to a slider - will be more useful.